src/blockstore/transaction.ts

```diff
@@ -137,6 +141,27 @@ export class BaseBlockstore implements BlockFetcher {
     return { t, meta: done };
   }
 
+  openTransaction(opts: CarTransactionOpts = { add: true, noLoader: false }): CarTransaction {
+    return new CarTransaction(this, opts);
+  }
+
+  async commitTransaction<M extends TransactionMeta>(
+    t: CarTransaction,
+    done: M,
+    opts: CarTransactionOpts,
+  ): Promise<TransactionWrapper<M>> {
+    if (!this.loader) throw this.logger.Error().Msg("loader required to commit").AsError();
+    const cars = await this.loader?.commit<M>(t, done, opts);
+    if (this.ebOpts.autoCompact && this.loader.carLog.length > this.ebOpts.autoCompact) {
+      setTimeout(() => void this.compact(), 10);
```

Comment:
this timeout is copied from the commit() function below. It's only every 100 writes, so this is just to delay after the current write finishes